x-pack/plugins/case/server/routes/api/utils.ts

```diff
@@ -191,18 +326,41 @@ export const sortToSnake = (sortField: string): SortFieldCase => {
 
 export const escapeHatch = schema.object({}, { unknowns: 'allow' });
 
-export const isUserContext = (context: CommentRequest): context is CommentRequestUserType => {
+/**
+ * A type narrowing function for user comments. Exporting so integration tests can use it.
+ */
+export const isCommentRequestTypeUser = (
+  context: CommentRequest
+): context is CommentRequestUserType => {
   return context.type === CommentType.user;
 };
 
-export const isAlertContext = (context: CommentRequest): context is CommentRequestAlertType => {
-  return context.type === CommentType.alert;
+/**
+ * A type narrowing function for alert comments. Exporting so integration tests can use it.
+ */
+export const isCommentRequestTypeAlertOrGenAlert = (
+  context: CommentRequest
+): context is CommentRequestAlertType => {
+  return context.type === CommentType.alert || context.type === CommentType.generatedAlert;
+};
+
+/**
+ * This is used to test if the posted comment is an generated alert. A generated alert will have one or many alerts.
+ * An alert is essentially an object with a _id field. This differs from a regular attached alert because the _id is
```

Comment:
nit: generated alerts do not have an `_id` field any more. Fields of `alert` and `generated_alert` is the same.

Reply:
Yep thanks.